resolver/src/hosts.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashMap;
+use std::fs;
+use std::time::{Duration, SystemTime};
+use std::io::{BufRead, BufReader};
+use std::fs::File;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::sync::Mutex;
+
+lazy_static! {
+    static ref CACHE_MAX_AGE: Duration = Duration::new(5, 0);
+    static ref HOSTS: Mutex<Hosts> = Mutex::new(Hosts::new());
+}
+
+#[derive(Debug)]
+pub struct Hosts {
+    by_name: HashMap<String, Vec<IpAddr>>,
+    expire: SystemTime,
```

Comment:
I get the desire to reload the file, but how valuable will this be in practice? It would simplify the structure to not reload quite a bit. And it's something that could be added later if need be.